lib/service/modules/groupStore.js

```diff
@@ -0,0 +1,127 @@
+/**
+ * Copyright (c) 2016 Kinvey Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+const BaseStore = require('./baseStore');
+const BASE_ROUTE = 'group';
+
+// Initialize the GroupStore; initializes to mastersecret, unless useUserContext = true
+function initGroupStore(appMetadata, requestContext, taskMetadata) {
+  class GroupStore extends BaseStore {
+    _buildUserRequest(options) {
+      return this._buildKinveyRequest(BASE_ROUTE, null, options);
+    }
+
+    _makeUserRequest(requestOptions, callback) {
+      if (this._taskMetadata.objectName === 'group' && (this._useBl === true || this._useUserContext === true)) {
+        const error = new Error('GroupStoreError');
+        error.description = 'Not Allowed';
+        error.debug = 'Recursive requests to the group store from the group store cannot use user credentials or use Bl';
+        return callback(error);
+      }
+      return this._makeRequest(requestOptions, callback);
+    }
+
+    create(group, callback) {
+      if (!group || typeof group === 'function') {
+        callback = group;
+        group = null;
+
+        const error = new Error('GroupStoreError');
+        error.description = 'Bad Request';
+        error.debug = 'A group entity must be supplied';
+        return callback(error);
+      }
+
+      const requestOptions = this._buildUserRequest({ useAppSecret: true });
```

Comment:
Creating groups does not use app secret credentials, it either uses user context or mastersecret.  So the options object here can be removed.